bitswap.go

```diff
@@ -266,18 +266,21 @@ func (bs *Bitswap) GetBlocks(ctx context.Context, keys []cid.Cid) (<-chan blocks
 				case <-ctx.Done():
 					return
 				}
+			case <-findProvsDelay.C:
+				// NB: Optimization. Assumes that providers of key[0] are likely to
+				// be able to provide for all keys. This currently holds true in most
+				// every situation. Later, this assumption may not hold as true.
+				bs.findKeys <- &blockRequest{
```

Comment:
Should we select on `ctx.Done()` here just in case?

Reply:
Good point. I ended up just reworking this patch to move sending the findProvs request to the main select. This *also* means we can abort this if we end up getting the block before the send completes (should help under heavy load).